raiden/tasks.py

```diff
@@ -134,6 +135,78 @@ def stop_async(self):
         self.stop_event.set(True)
 
 
+class HealthcheckTask(Task):
+    """ Task for checking if all of our open channels are healthy """
+
+    def __init__(
+            self,
+            raiden,
+            send_ping_time,
+            max_unresponsive_time,
+            sleep_time=DEFAULT_HEALTHCHECK_POLL_TIMEOUT):
+        """ Initialize a HealthcheckTask that will monitor open channels for
+             responsiveness.
+
+             :param raiden RaidenService: The Raiden service which will give us
+                                          access to the protocol object and to
+                                          the asset manager
+             :param int sleep_time: Time in seconds between each healthcheck task
+             :param int send_ping_time: Time in seconds after not having received
+                                        a message from an address at which to send
+                                        a Ping.
+             :param int max_unresponsive_time: Time in seconds after not having received
+                                               a message from an address at which it
+                                               should be deleted.
+         """
+        super(HealthcheckTask, self).__init__()
+
+        self.protocol = raiden.protocol
+        self.raiden = raiden
+
+        self.stop_event = AsyncResult()
+        self.sleep_time = sleep_time
+        self.send_ping_time = send_ping_time
+        self.max_unresponsive_time = max_unresponsive_time
+
+    def _run(self):  # pylint: disable=method-hidden
+        stop = None
+        while stop is None:
+            keys_to_remove = []
+            for key, queue in self.protocol.address_queue.iteritems():
+                receiver_address = key[0]
+                asset_address = key[1]
+                try:
```

Comment:
Hmm yes ... I was thinking of blocking for a tiny bit there, but it makes no sense right? We don't need to wait at all there, just check if there is anything in the queue or not at the time the task runs.